File: examples/chat/producer-rust/src/events/disconnected.rs
use crate::{identification, producer::Control, protocol, Context};
use uuid::Uuid;

type BroadcastMessage = (Vec<Uuid>, protocol::Events::Message);

#[allow(unused_variables)]
pub async fn emit(
    uuid: Uuid,
    context: &mut Context,
    filter: identification::Filter,
    control: &Control,
) -> Result<BroadcastMessage, String> {
    Err(String::from(
        "Event emitter \"disconnected\" isn't implemented",
    ))
}

File: tests/workflow/producer/typescript/src/producer/implementation/events/disconnected.ts
import { Producer, Identification, Filter, Context } from "./index";
import { emit } from "../../events/disconnected";

export function handler(
	consumer: Identification,
	filter: Filter,
	context: Context,
	producer: Producer
): Promise<void> {
	return emit(consumer, filter, context, producer);
}
